recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+            self.requires(f"cpython/{self.options.python_version}")
+            if is_msvc(self):
+                # Required to create import .lib for building extension module.
+                self.options["cpython"].shared = True
+            self.requires("pybind11/2.10.0")
+
+    def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support MacOS at this time"
+            )
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.21.7")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["OPENASSETIO_ENABLE_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+        libcxx = self.settings.get_safe("compiler.libcxx")
+        if libcxx is not None:
+            if libcxx == "libstdc++11":
+                tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = True
+            else:
+                tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = False
+
+        if self.options.without_python:
+            tc.variables["OPENASSETIO_ENABLE_PYTHON"] = False
+        else:
+            tc.variables["OPENASSETIO_ENABLE_PYTHON"] = True
+            tc.variables["Python_EXECUTABLE"] = self._python_exe
+            if is_msvc(self):
+                tc.variables["Python_LIBRARY"] = self._python_windows_lib
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    @property
+    def _python_exe(self):
+        return pathlib.Path(self.deps_user_info["cpython"].python).as_posix()
+
+    @property
+    def _python_windows_lib(self):
+        pth = pathlib.Path(
+            self.deps_cpp_info["cpython"].rootpath,
+            self.dependencies["cpython"].cpp_info.components["embed"].libdirs[0],
+            self.dependencies["cpython"].cpp_info.components["embed"].libs[0])
+        pth = pth.with_suffix(".lib")
+        return pth.as_posix()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package_id(self):
+        if not self.options.without_python:
+            self.info.requires["cpython"].minor_mode()
+
```

Comment:
```suggestion
```

It can not be done. Otherwise, it will break ConanCenterIndex, as your package will require a package which is not available due different package ID

Reply:
Ah interesting, I thought `minor_mode` meant that the package ID hash of the `openassetio` recipe would incorporate the minor version of `cpython`?  

That would be good to enforce, since, anonyingly, Python has ABI breakages in-between minor versions.  Is there a way to do this?

Reply:
Thinking again, there are few packages using minor version, so we can keep it. What we can not do, is changing cpython recipe to use minor mode by default. However, we see that's required, since Python breaks its ABI on minor versions. As result, people who want to use another version of python with this package, will need to building everything again.